src/tests/ftest/util/nvme_utils.py

```diff
@@ -44,19 +44,19 @@ def set_device_faulty(test, dmg, server, uuid, pool=None, has_sys_xs=False, **kw
     Args:
         test (Test): avocado test class
         dmg (DmgCommand): a DmgCommand class instance
-        server (NodeSet): host on which to issue the dmg storage set nvme-faulty
+        server (NodeSet): host on which to issue the dmg storage set nvme-faulty. Must be one host.
         uuid (str): the device UUID
         pool (TestPool, optional): pool used to wait for rebuild to start/complete if specified.
             Defaults to None.
         has_sys_xs (bool, optional): the device's has_sys_xs property value. Defaults to False.
         kwargs (dict, optional): named arguments to pass to the DmgCommand.storage_set_faulty.
 
     Returns:
-        dict: the json response from the dmg storage set-faulty command.
-
+        dict: the json response from the dmg storage set-faulty command. None if has_sys_xs is True.
```

Comment:
It's always None if has_sys_xs is True? In that case, what's the point in calling this function if it's just expected to fail every time?

Reply:
We want to call the function to set the device faulty.  The behavior afterward is what differs with a sys_xs device.  In that case we need to tell the test that we expect specific ranks to be stopped/excluded.  Otherwise we need to register a cleanup step to reset the faulty device - which we can do because the ranks are still up with a non sys_xs device.

Technically this method doesn't need to return the response as its stored in the `dmg.result`.  I had made that change in https://github.com/daos-stack/daos/pull/14389/files.

Reply:
I think it's better practice to use the returned response and not the `dmg.result`

Reply:
So if `has_sys_xs is True` this will call `dmg.storage_set_faulty`, which may or may not work. But we ignore the failure.

So
`if has_sys_xs is False` -> call `dmg.storage_set_faulty` and fail the test if it fails. But if it works return the response
`if has_sys_xs is True` -> call `dmg.storage_set_faulty` and don't care if it works or not. But if it did work return the response, even though the docstring says it will return `None`

Reply:
> I think it's better practice to use the returned response and not the `dmg.result`

I agree, there are problems with `dmg.result` in terms of knowing when it was last updated.  I just wasn't sure if your concern was with the None return value.  Ideally ExecutableCommand.run would use run_local/run_remote and we'd always have an object to return, but that's way out of the scope of this PR.

Reply:
> dict: the json response from the dmg storage set-faulty command. None if has_sys_xs is True.

If this is correct, then the code should verify that `storage_set_faulty` always fails when `has_sys_xs is True`